compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -746,6 +746,17 @@ trait Checking {
         em"Implementation restriction: ${path.tpe.classSymbol} is not a valid prefix " +
           "for a wildcard export, as it is a package.", path.srcPos)
 
+  /** Check that module `sym` does not clash with a class of the same name
+   *  that is concurrently compiled in another source file.
+   */
+  def checkNoModuleClash(sym: Symbol)(using Context): Unit =
+    if sym.effectiveOwner.is(Package)
+       && sym.owner.info.member(sym.name.moduleClassName).symbol.isAbsent()
```

Comment:
I don't understand the logic here. It seems that we always call `checkNoModuleClash` with a module symbol. If that's the case, then shouldn't this symbol always be non-absent? Or is it that this symbol was marked absent when we added the clashing class's symbol as a member?

Reply:
Yes, the latter.